### PR TITLE
Added meshkit style error in handlers

### LINKED ISSUE #3303: [Chore] Update filter, pattern, and oam handlers to use new Meshkit Errors
Note: This is a child issue of #3275 
#### Current Behavior
Meshery Server (and its other components, like Adapters) use a centralized library, MeshKit, to provide uniform utilities (e.g. raising and logging errors). Currently, Meshery Server uses a now antiquated style of error logging with logrus.

#### Desired Behavior
Ideally, all use of logrus is replaced with use of MeshKit-style error logging.

#### Implementation
Meshery Handlers needs to be:
 - Updated to use meshkit errors. 
 - Switch to handler Log instead of `logrus`

Handlers to update:

- [ ] meshery filter handler
- [ ]  meshery pattern handler
- [x]  oam handler

For reference, you can checkout  the [`grafana handler`](https://github.com/layer5io/meshery/blob/master/handlers/grafana_handlers.go) and [`prometheus handler`](https://github.com/layer5io/meshery/blob/master/handlers/prometheus_handlers.go) or the [docs](https://docs.meshery.io/project/contributing-error).
Define all Error functions for all the handlers to be written in the `handlers/error.go` file.

#### Mockups

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

**Description**

Added the Meshkit style error in the

1. Meshery Filter handler
2. Meshery Pattern handler

This PR fixes #3303 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: handlers/error.go [hunk 1]
```diff
@@ -322,3 +338,67 @@ func ErrInvalidRequestObject(fields ...string) error {
 func ErrChangeK8sContext(err error) error {
 	return errors.New(ErrCreateDirCode, errors.Alert, []string{"Error changing context"}, []string{err.Error()}, []string{"Context Name might be invalid or not present in the uploaded kubeconfig"}, []string{"Check the context name, if the context name is correct and is present in the kubeconfig then try uploading the kubeconfig again"})
 }
+
+func ErrGetFilter(err error) error {
+	return errors.New(ErrGetFilterCode, errors.Alert, []string{"Error failed to get filter"}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrSaveFilter(err error) error {
+	return errors.New(ErrSaveFilterCode, errors.Alert, []string{"Error failed to save filter"}, []string{err.Error()}, []string{}, []string{})
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply handlers/error.go:347]: Look to the Error Code Guide in docs.meshery.io for examples, @sayantan1413

**Review by utkarsh-pro — DISMISSED** [review]
LGTM. Some items by @leecalcote needs to be addressed.

**Comment by sayantan1413** [comment]
Yes @utkarsh-pro I will do those changes soon.

File: handlers/error.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sayantan1413) [reply handlers/error.go:347]: Sure I will add the probable cause and suggested remediation

**Comment by navendu-pottekkat** [comment]
@sayantan1413 Is this PR ready for a review?

**Review by utkarsh-pro — APPROVED** [review]
LGTM. @leecalcote kindly merge if your comments regarding "probable cause and suggested remediation".

**Review by bariqhibat — CHANGES_REQUESTED** [review]
Hey @sayantan1413, any updates? We're left with suggested remedition and probable cause, then I think we're good to go (?)

**Comment by sayantan1413** [comment]
Hey, @bariqhibat I have already completed writing those.

**Review by bariqhibat — APPROVED** [review]
Looks good

**Review by leecalcote — APPROVED** [review]

**Pull request merged**